build: Remove CMAKE_SKIP_BUILD_RPATH and SKIP_BUILD_RPATH settings

151henry151 opened the pull request:
Remove CMAKE_SKIP_BUILD_RPATH and SKIP_BUILD_RPATH settings that are no longer needed after reordering the Guix build script to perform binary checks after installation.

This PR also removes the unused CMake maintenance targets (`check-security` and `check-symbols`) and updates the Guix security checks to include binaries in the `libexec/` directory (added in PR #31679).

hebasto reviewed (commented):
> This is my first attempt at contributing to bitcoin core so please, provide me any feedback that you might have. Thanks!

Thank you for your interest in contributing to this project!

The changes you are referring to as completed, i.e. "reordering Guix script commands to perform binary checks after the installation step", have not actually been done yet. See [`contrib/guix/libexec/build.sh`](https://github.com/bitcoin/bitcoin/blob/master/contrib/guix/libexec/build.sh), from line 251 onwards.

hebasto reviewed (commented):
In a02723427f5f0b142a6ef97b92df093e00ad6004, all checks are still performed on the binaries in the build tree, whereas they are supposed to be performed on the binaries in the installation location.

151henry151 commented:
> ### Conflicts
> 
> Reviewers, this pull request conflicts with the following ones:
> 
>     * [#32380](https://github.com/bitcoin/bitcoin/pull/32380) (Modernize use of UTF-8 in Windows code by hebasto)
> 
The conflict resolves easily by keeping both changes.

151henry151 commented:
> In [a027234](https://github.com/bitcoin/bitcoin/commit/a02723427f5f0b142a6ef97b92df093e00ad6004), all checks are still performed on the binaries in the build tree, whereas they are supposed to be performed on the binaries in the installation location.

Thanks for the feedback.

The previous change moved checks after installation but was still using cmake --build targets which check build tree binaries. I've now fixed this to directly call the security and symbol check scripts on the installed binaries in ${INSTALLPATH}/bin/, ensuring checks are performed on the final installed binaries with proper RPATHs.

>When a developer tests changes in the Guix scripts locally, they usually posts the resulting binary hashes. For example, as in https://github.com/bitcoin/bitcoin/pull/33217#pullrequestreview-3142128074.

I'm currently running a full Guix build locally to generate the binary hashes. The build is in progress and I'll post the resulting hashes once it completes.

151henry151 commented:
My Guix build resulting binary hashes:

`75d4eb10b95a64c595ee1ddf31d3c1e0086958a1ae3d64f9e2bbc4930cd88366  dist-archive/bitcoin-e922e27a2ecd.tar.gz
f7d2d6cd820561229ab5fa476c06484ced81a1fa8b90d2fbbd7a82fa915aa956  x86_64-linux-gnu/bitcoin-e922e27a2ecd-x86_64-linux-gnu-debug.tar.gz
df24c2269cf96dbf011692fbc0b21dab7786abd43dbb82e4351f0117e95169e5  x86_64-linux-gnu/bitcoin-e922e27a2ecd-x86_64-linux-gnu.tar.gz`

janb84 commented on CMakeLists.txt:
```diff
@@ -618,17 +618,16 @@ if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.29)
-# TODO: The `CMAKE_SKIP_BUILD_RPATH` variable setting can be deleted
-#       in the future after reordering Guix script commands to
-#       perform binary checks after the installation step.
+# TODO: The `CMAKE_SKIP_BUILD_RPATH` variable setting has been removed
```
```suggestion

```

If the TODO is solved than there is no  more TO DO. Please remove the todo or reword it what is still left to do.

151henry151 replied:
I clicked commit suggestion too quickly -- It looks a bit like a comment was cut in half there. Would we want to remove the additional lines of the TODO as well, or rephrase them as comments?

151henry151 reviewed (commented):
Would it be better to leave the comments there and simply remove the TODO tag? It looks a bit like a comment was cut in half there, but perhaps adding back in the 
```The `CMAKE_SKIP_BUILD_RPATH` variable setting has been removed```
Would be better than removing the rest of that information. What do you think?

151henry151 commented:
Is there anything further required here? I understand that review for pull requests can take a long time (especially for non-critical things such as this one) but just wanted to check in and see if there's further action needed from me, or anything about this pull request that I could improve.

151henry151 commented on src/CMakeLists.txt:
```diff
@@ -410,14 +410,13 @@ if(BUILD_UTIL_CHAINSTATE)
   # - https://github.com/bitcoin/bitcoin/pull/30312#issuecomment-2191235833
   set_target_properties(bitcoin-chainstate PROPERTIES
-    SKIP_BUILD_RPATH OFF
   )
```
I think these have been squashed correctly to address these clumsy mistakes and that the final three commits in the pull request reflect this.

151henry151 commented on src/CMakeLists.txt:
```diff
@@ -410,14 +410,13 @@ if(BUILD_UTIL_CHAINSTATE)
-  #       perform binary checks after the installation step.
+  # TODO: The `SKIP_BUILD_RPATH` property setting has been removed
+  #       after reordering Guix script commands to perform binary checks
+  #       after the installation step.
```
I believe this has already been addressed by squashing the relevant commits.

151henry151 commented:
> Please squash your commits according to https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#squashing-commits

I'm still learning, but I think I did the squashing correctly now, hopefully! Thanks for your patience and hope I can be of some value to the project :)

purpleKarrot commented:
Code review ACK. I very much welcome the reduced complexity in the CMake configuration. Thanks for working on this!

I haven't tested actually running the scripts yet.

janb84 commented on contrib/guix/libexec/build.sh:
```diff
@@ -282,6 +277,14 @@ mkdir -p "$DISTSRC"
+    # Perform basic security checks on INSTALLED executables.
+    # These checks now happen after installation, so RPATHs point to final locations.
+    echo "Checking binary security on installed executables..."
+    python3 "${DISTSRC}/contrib/guix/security-check.py" "${INSTALLPATH}/bin/"*
```
NIT: This only covers 1 of the 2 binary locations, nowadays we also have `libexec` should this not also be checked ?

151henry151 replied:
Thanks for the suggestion! I think I've addressed this correctly in [07027af](https://github.com/bitcoin/bitcoin/pull/33247/commits/07027afa155f8ec9715c761d80630b31723c2c32) -- built for linux successfully; here's my GUIX build hashes:

```8890962245e7c04cba900e0c5b6c57e2d9b3779f08d7bb6f0f1ec8baee9a1425  dist-archive/bitcoin-07027afa155f.tar.gz
4ed2d3dfe075bf5f98042bfe2165d858a4913b6933c8098e52cf607fbe76c84b  x86_64-linux-gnu/bitcoin-07027afa155f-x86_64-linux-gnu-debug.tar.gz
7b11bd93f10a37554fffd6128ede1a47585348ad8d12687b39f7e7d4f0b144f8  x86_64-linux-gnu/bitcoin-07027afa155f-x86_64-linux-gnu.tar.gz

janb84 reviewed (commented):
Concept ACK a9b211540b79832fb4e4f870d2f770cf03d35b11

This PR reorders the binary checks to after the installation step. And removes the code paths that are made superfluous by the reorg .

Open NIT/question on `libexec` inclusion.

Guix builds and the security tests runs now after install: ✅
<details>

This PR:
```shell
[101%] Built target test_bitcoin
-- Install configuration: "RelWithDebInfo"
-- Installing: /distsrc-base/distsrc-a9b211540b79-riscv64-linux-gnu/installed/bitcoin-a9b211540b79/bin/bitcoin-wallet
-- Installing: /distsrc-base/distsrc-a9b211540b79-riscv64-linux-gnu/installed/bitcoin-a9b211540b79/share/man/man1/bitcoin-wallet.1
-- Installing: /distsrc-base/distsrc-a9b211540b79-riscv64-linux-gnu/installed/bitcoin-a9b211540b79/bin/bitcoin
-- Installing: /distsrc-base/distsrc-a9b211540b79-riscv64-linux-gnu/installed/bitcoin-a9b211540b79/bin/bitcoind
-- Installing: /distsrc-base/distsrc-a9b211540b79-riscv64-linux-gnu/installed/bitcoin-a9b211540b79/share/man/man1/bitcoind.1
-- Installing: /distsrc-base/distsrc-a9b211540b79-riscv64-linux-gnu/installed/bitcoin-a9b211540b79/libexec/bitcoin-node
-- Installing: /distsrc-base/distsrc-a9b211540b79-riscv64-linux-gnu/installed/bitcoin-a9b211540b79/bin/bitcoin-cli
-- Installing: /distsrc-base/distsrc-a9b211540b79-riscv64-linux-gnu/installed/bitcoin-a9b211540b79/share/man/man1/bitcoin-cli.1
-- Installing: /distsrc-base/distsrc-a9b211540b79-riscv64-linux-gnu/installed/bitcoin-a9b211540b79/bin/bitcoin-tx
-- Installing: /distsrc-base/distsrc-a9b211540b79-riscv64-linux-gnu/installed/bitcoin-a9b211540b79/share/man/man1/bitcoin-tx.1
-- Installing: /distsrc-base/distsrc-a9b211540b79-riscv64-linux-gnu/installed/bitcoin-a9b211540b79/bin/bitcoin-util
-- Installing: /distsrc-base/distsrc-a9b211540b79-riscv64-linux-gnu/installed/bitcoin-a9b211540b79/share/man/man1/bitcoin-util.1
-- Installing: /distsrc-base/distsrc-a9b211540b79-riscv64-linux-gnu/installed/bitcoin-a9b211540b79/bin/bitcoin-qt
-- Set runtime path of "/distsrc-base/distsrc-a9b211540b79-riscv64-linux-gnu/installed/bitcoin-a9b211540b79/bin/bitcoin-qt" to ""
-- Installing: /distsrc-base/distsrc-a9b211540b79-riscv64-linux-gnu/installed/bitcoin-a9b211540b79/share/man/man1/bitcoin-qt.1
-- Installing: /distsrc-base/distsrc-a9b211540b79-riscv64-linux-gnu/installed/bitcoin-a9b211540b79/libexec/bitcoin-gui
-- Set runtime path of "/distsrc-base/distsrc-a9b211540b79-riscv64-linux-gnu/installed/bitcoin-a9b211540b79/libexec/bitcoin-gui" to ""
-- Installing: /distsrc-base/distsrc-a9b211540b79-riscv64-linux-gnu/installed/bitcoin-a9b211540b79/libexec/test_bitcoin
Checking binary security on installed executables...
Running symbol and dynamic library checks on installed executables...
```

V30.0rc1:

```shell
[100%] Built target bitcoin-gui
Checking binary security...
[100%] Built target check-security
[  2%] Built target test_util
[  2%] Built target crc32c
[  8%] Built target leveldb
[  9%] Built target minisketch
[...]
[100%] Built target bitcoin-gui_autogen
[100%] Built target bitcoin-gui
Running symbol and dynamic library checks...
[100%] Built target check-symbols
-- Install configuration: "RelWithDebInfo"
-- Installing: /distsrc-base/distsrc-30.0rc1-riscv64-linux-gnu/installed/bitcoin-30.0rc1/bin/bitcoin-wallet
-- Installing: /distsrc-base/distsrc-30.0rc1-riscv64-linux-gnu/installed/bitcoin-30.0rc1/share/man/man1/bitcoin-wallet.1
-- Installing: /distsrc-base/distsrc-30.0rc1-riscv64-linux-gnu/installed/bitcoin-30.0rc1/bin/bitcoin
-- Installing: /distsrc-base/distsrc-30.0rc1-riscv64-linux-gnu/installed/bitcoin-30.0rc1/bin/bitcoind
-- Installing: /distsrc-base/distsrc-30.0rc1-riscv64-linux-gnu/installed/bitcoin-30.0rc1/share/man/man1/bitcoind.1
-- Installing: /distsrc-base/distsrc-30.0rc1-riscv64-linux-gnu/installed/bitcoin-30.0rc1/libexec/bitcoin-node
```

</details>

janb84 reviewed (commented):
Concept ACK 07027afa155f8ec9715c761d80630b31723c2c32

NIT: The separate commit has no additional value, please squash them. 

Checked that 
- the bash script halts if python script exits with code 1 ✅
- All the binaries are checked by the python script ✅

<details> 
edited the python script to check that all the binaries are being checked (that the build script calls the security python check correctly)

```shell

Checking binary security on installed executables...

Checking /distsrc-base/distsrc-26c40cec0448-riscu64-linux-gnu/installed/bitcoin-26c40cec0448/bin/bitcoin (ELF RISCU)
Checking /distsrc-base/distsrc-26c40cec0448-riscu64-linux-gnu/installed/bitcoin-26c40cec0448/bin/bitcoin-cli (ELF RISCV)
Checking /distsrc-base/distsrc-26c40cec0448-riscu64-linux-gnu/installed/bitcoin-26c40cec0448/bin/bitcoin-qt (ELF RISCV)
Checking /distsrc-base/distsrc-26c40cec0448-riscu64-linux-gnu/installed/bitcoin-26c40cec0448/bin/bitcoin-tx (ELF RISCV)
Checking /distsrc-base/distsrc-26c40cec0448-riscu64-linux-gnu/installed/bitcoin-26c40cec0448/bin/bitcoin-util (ELF RISCV)
Checking /distsrc-base/distsrc-26c40cec0448-riscu64-linux-gnu/installed/bitcoin-26c40cec0448/bin/bitcoin-wallet (ELF RISCV)
Checking /distsrc-base/distsrc-26c40cec0448-riscu64-linux-gnu/installed/bitcoin-26c40cec0448/bin/bitcoind (ELF RISCV)
Checking /distsrc-base/distsrc-26c40cec0448-riscu64-linux-gnu/installed/bitcoin-26c40cec0448/libexec/bitcoin-gui (ELF RISCU)
Checking /distsrc-base/distsrc-26c40cec0448-riscu64-linux-gnu/installed/bitcoin-26c40cec0448/libexec/bitcoin-node (ELF RISCU)
Checking /distsrc-base/distsrc-26c40cec0448-riscu64-linux-gnu/installed/bitcoin-26c40cec0448/libexec/test_bitcoin (ELF RISCU)

```

hebasto commented on CMakeLists.txt:
```diff
@@ -618,17 +618,10 @@ if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.29)
   set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
 else()
-  set(CMAKE_SKIP_BUILD_RPATH TRUE)
   set(CMAKE_SKIP_INSTALL_RPATH TRUE)
```
It's outside the scope of this PR, but `CMAKE_SKIP_INSTALL_RPATH` should be set in the Guix script, keeping the build system free of hardcoded behaviour.

hebasto reviewed (commented):
Approach ACK 07027afa155f8ec9715c761d80630b31723c2c32.

The comment "`TODO: The `CMAKE_SKIP_BUILD_RPATH` variable setting can be deleted...`" does not imply that anything is deprecated. Could you please update the PR description and commit message accordingly?

janb84 reviewed (commented):
ACK 9c13be9c45cad19d9db78e318b1e8376d56d6ec5

Thanks for squashing your commits!

My Guix build checksums: 
**Commit:** `9c13be9c45ca`

```shell
 22ad5809ed8d8399b0c7960e8c57cbaae5220b2667650262a7ec98a25eaf671f  guix-build-9c13be9c45ca/output/aarch64-linux-gnu/SHA256SUMS.part
 60b2048157462a5bd7ae00fa60ec8f1d59d84857d55125b0eafa98fad4aa780e  guix-build-9c13be9c45ca/output/aarch64-linux-gnu/bitcoin-9c13be9c45ca-aarch64-linux-gnu-debug.tar.gz
 7a7280f6b1221a62e9e8591365431f2cc3a0fe111abe76799df774e5b1e82207  guix-build-9c13be9c45ca/output/aarch64-linux-gnu/bitcoin-9c13be9c45ca-aarch64-linux-gnu.tar.gz
 24c34eb9d583c6dc1f9adddf7d3ebd6e717c278cd51b454c767c5f6bf2c78ec0  guix-build-9c13be9c45ca/output/arm-linux-gnueabihf/SHA256SUMS.part
 cdd558dab24381ac72ac805ad5f8715bebebe291e3ba7fc7c71070364a18d066  guix-build-9c13be9c45ca/output/arm-linux-gnueabihf/bitcoin-9c13be9c45ca-arm-linux-gnueabihf-debug.tar.gz
 fdba945dd73d5cf2ba6413ec499c459708eda217d6c64d89b1ecf8300721c88a  guix-build-9c13be9c45ca/output/arm-linux-gnueabihf/bitcoin-9c13be9c45ca-arm-linux-gnueabihf.tar.gz
 edf1f03a79da8eda440e45625e0617e486847868215cc51ed6222900df94a8e9  guix-build-9c13be9c45ca/output/arm64-apple-darwin/SHA256SUMS.part
 b6ecf45d5fff2668a6f6d127da7e37c22680e06855a19b89f7b719ad436a3cdf  guix-build-9c13be9c45ca/output/arm64-apple-darwin/bitcoin-9c13be9c45ca-arm64-apple-darwin-codesigning.tar.gz
 fbcdc8bb52facac2e0f686c874f9a89c3850564deb106bcc2b2030673e48464f  guix-build-9c13be9c45ca/output/arm64-apple-darwin/bitcoin-9c13be9c45ca-arm64-apple-darwin-unsigned.tar.gz
 692a571078b0ee482fb7559805087b030302d368439088c7d6d2c68e1b09bac2  guix-build-9c13be9c45ca/output/arm64-apple-darwin/bitcoin-9c13be9c45ca-arm64-apple-darwin-unsigned.zip
 03bb2e49ce986ddb620d7672c7c30168880c45802008c81e9eb6f878f4081d1c  guix-build-9c13be9c45ca/output/dist-archive/bitcoin-9c13be9c45ca.tar.gz
 4d1c2db2282d76393f0024ce8f39a82eeb5892df8f7196b333458e9e4cdca719  guix-build-9c13be9c45ca/output/powerpc64-linux-gnu/SHA256SUMS.part
 5d1c08b16f990ae7f74c6503b08aeb378080110af5c3a92507e7285694da81fd  guix-build-9c13be9c45ca/output/powerpc64-linux-gnu/bitcoin-9c13be9c45ca-powerpc64-linux-gnu-debug.tar.gz
 72609c0794620a5977c49949202d59222c0723eaa9eabd0e37f183e902ff1929  guix-build-9c13be9c45ca/output/powerpc64-linux-gnu/bitcoin-9c13be9c45ca-powerpc64-linux-gnu.tar.gz
 b1354bb7e032157a3a68eb39f15a5a0aefb3fb071cfac6d2d13ce87ce22c0f52  guix-build-9c13be9c45ca/output/riscv64-linux-gnu/SHA256SUMS.part
 918d66e4d0f000c082f7a9bb62aa4606df612d13adefc1270e16d6d4351f8da8  guix-build-9c13be9c45ca/output/riscv64-linux-gnu/bitcoin-9c13be9c45ca-riscv64-linux-gnu-debug.tar.gz
 95fc99008bfcf913d411234fc63fb4abb433e57d9205617a5a8786c2ab936ad9  guix-build-9c13be9c45ca/output/riscv64-linux-gnu/bitcoin-9c13be9c45ca-riscv64-linux-gnu.tar.gz
 c4f3ee6367485a4a0f37495d93bb19a1acc6fe6e485c47510cf7fb661d4dc002  guix-build-9c13be9c45ca/output/x86_64-apple-darwin/SHA256SUMS.part
 eb147cdac24259b279b8999745cb89bd17762df8261bb678e06148cc30348e06  guix-build-9c13be9c45ca/output/x86_64-apple-darwin/bitcoin-9c13be9c45ca-x86_64-apple-darwin-codesigning.tar.gz
 1856a8e2f4f5d06db427f3f66066b01d411eeb6b87980b878f374460e13502dc  guix-build-9c13be9c45ca/output/x86_64-apple-darwin/bitcoin-9c13be9c45ca-x86_64-apple-darwin-unsigned.tar.gz
 3f87e1e08ff5c6755765ec85160f31c25cac3351dc49a4d52af08ce894315490  guix-build-9c13be9c45ca/output/x86_64-apple-darwin/bitcoin-9c13be9c45ca-x86_64-apple-darwin-unsigned.zip
 98727d830043e223bc75d7e756412a9e56b995af1c3db3a963fee3f5ce92a6ed  guix-build-9c13be9c45ca/output/x86_64-linux-gnu/SHA256SUMS.part
 d781e85479e6c0c62eedf4b59db5c32c5342ed9dfe49daa423b71cfcc0906bf2  guix-build-9c13be9c45ca/output/x86_64-linux-gnu/bitcoin-9c13be9c45ca-x86_64-linux-gnu-debug.tar.gz
 799c230ba61826a9ebe39088d074d4c5608266e469da9356044ebbf50c60b28e  guix-build-9c13be9c45ca/output/x86_64-linux-gnu/bitcoin-9c13be9c45ca-x86_64-linux-gnu.tar.gz
 916d6a93a5b0f174319afb351eaf3dfa5edcf6648fec3b0b23250bc6be6f9395  guix-build-9c13be9c45ca/output/x86_64-w64-mingw32/SHA256SUMS.part
 0a7413e714a7bfd7dbc327ef9e6f62e9472f3214d40c678c62f818aadb93343d  guix-build-9c13be9c45ca/output/x86_64-w64-mingw32/bitcoin-9c13be9c45ca-win64-codesigning.tar.gz
 d26b0ff872e8e5db1b5f87468e8d66be0114a3088ef0fca740dd7f629a966738  guix-build-9c13be9c45ca/output/x86_64-w64-mingw32/bitcoin-9c13be9c45ca-win64-debug.zip
 a915942305a64eb76bb48afef01c4a8532e34ec2434239a3cc558e279dcf952b  guix-build-9c13be9c45ca/output/x86_64-w64-mingw32/bitcoin-9c13be9c45ca-win64-setup-unsigned.exe
 461d043ae6a8b0208eaa492a23f9d9ec8f8eda9773360d9e4c5a200f873ed26b  guix-build-9c13be9c45ca/output/x86_64-w64-mingw32/bitcoin-9c13be9c45ca-win64-unsigned.zip
```

hebasto commented on contrib/guix/libexec/build.sh:
```diff
@@ -282,6 +277,14 @@ mkdir -p "$DISTSRC"
             ;;
     esac
 
+    # Perform basic security checks on INSTALLED executables.
```
Is there any particular reason to capitalize "installed" here?

hebasto commented on contrib/guix/libexec/build.sh:
```diff
@@ -282,6 +277,14 @@ mkdir -p "$DISTSRC"
     esac
 
+    # Perform basic security checks on INSTALLED executables.
+    # These checks now happen after installation, so RPATHs point to final locations.
```
This comment doesn't seem correct to me, since we check that installed binaries have no `RUNPATH` or `RPATH` set. Maybe drop this line?

hebasto reviewed (commented):
I've tested 9c13be9c45cad19d9db78e318b1e8376d56d6ec5, including running with [`V=1`](https://github.com/bitcoin/bitcoin/blob/master/contrib/guix/README.md#recognized-environment-variables). It looks good.

Please amend the comments in `contrib/guix/libexec/build.sh` as noted above.

hebasto reviewed (commented):
My Guix build:
```
aarch64
ab375345e39248d2c61fe6fce1afeb7b95623cfc9db452a7cfbd5b6797bf9c16  guix-build-d55753626292/output/aarch64-linux-gnu/SHA256SUMS.part
85bd20c1b7076387d8efcb5ade58d29774f8114a712492a77315627d0db41094  guix-build-d55753626292/output/aarch64-linux-gnu/bitcoin-d55753626292-aarch64-linux-gnu-debug.tar.gz
f52e90c92d99b17c916b2c3ff2b57608597ca39ec7324ff5840550636cfd3a50  guix-build-d55753626292/output/aarch64-linux-gnu/bitcoin-d55753626292-aarch64-linux-gnu.tar.gz
9c8c935f51cc751bb7f92f918bcb7197518b906f103cc8c1ba5d8a92aa8d34ec  guix-build-d55753626292/output/arm-linux-gnueabihf/SHA256SUMS.part
2296587e0370c300eb289ac6fd8216de97ec2eb2516b5af3cea6d390ac7de478  guix-build-d55753626292/output/arm-linux-gnueabihf/bitcoin-d55753626292-arm-linux-gnueabihf-debug.tar.gz
4a352b62b9ab50f830c7df6450ff6510862dc867a265480855a28429fb080942  guix-build-d55753626292/output/arm-linux-gnueabihf/bitcoin-d55753626292-arm-linux-gnueabihf.tar.gz
be4ee4d187d92b9387e84e8e6208254a2c5bab175cac2a70e44bf2a078ba38e7  guix-build-d55753626292/output/arm64-apple-darwin/SHA256SUMS.part
06d67aa4b7a313de6da9a3f30cb75ec5047c9384e54aef0eb0de262d2ada71e6  guix-build-d55753626292/output/arm64-apple-darwin/bitcoin-d55753626292-arm64-apple-darwin-codesigning.tar.gz
ae1857537795adf11526d03f97185bb560269b2c9f01aac1e529c5795c59d10d  guix-build-d55753626292/output/arm64-apple-darwin/bitcoin-d55753626292-arm64-apple-darwin-unsigned.tar.gz
c7785d204823eededf39ae432c025ef4273305a28808b18f08fa8af04f9eed80  guix-build-d55753626292/output/arm64-apple-darwin/bitcoin-d55753626292-arm64-apple-darwin-unsigned.zip
7db1175dc6522b2bfdc555f4cf531b0d4b24a2cd8ebf122526fb812f5deee815  guix-build-d55753626292/output/dist-archive/bitcoin-d55753626292.tar.gz
e944b955363d82291dd61f89d5c145ff0e15c89769dfe4e94d7b305686578b7d  guix-build-d55753626292/output/powerpc64-linux-gnu/SHA256SUMS.part
f5783aaba0d7bd34d357bb125cc81570399873457276f402969d65ecaf67ef08  guix-build-d55753626292/output/powerpc64-linux-gnu/bitcoin-d55753626292-powerpc64-linux-gnu-debug.tar.gz
e357b89396ad99721becfb6092918a67e2f7fdb2a60e3e8b56bbc6662d1686d0  guix-build-d55753626292/output/powerpc64-linux-gnu/bitcoin-d55753626292-powerpc64-linux-gnu.tar.gz
58b0b34b6f1cfebdfac5b8e86dc70710644fbfc6c1f227b938a05a6267bdc108  guix-build-d55753626292/output/riscv64-linux-gnu/SHA256SUMS.part
745e1c1cc8dfef7cd72b9d50562acab5281237e198e34ca6d69273fa10bab1fa  guix-build-d55753626292/output/riscv64-linux-gnu/bitcoin-d55753626292-riscv64-linux-gnu-debug.tar.gz
1356ff14875a5828573d0b9dea35f8c147f011fb6c766dfcd3d11b1452c91b4a  guix-build-d55753626292/output/riscv64-linux-gnu/bitcoin-d55753626292-riscv64-linux-gnu.tar.gz
0f341adc5565f4142ba60adbafc191a7db3c2a46d09691a1d2057d8f883f60b3  guix-build-d55753626292/output/x86_64-apple-darwin/SHA256SUMS.part
01bf822e17e7c29dcf19915dc3ed86ec5a5e71b46ac08fa9d56c62070b5d8d8a  guix-build-d55753626292/output/x86_64-apple-darwin/bitcoin-d55753626292-x86_64-apple-darwin-codesigning.tar.gz
9178b67b056bbe462f080886409372eea8b7536be1cf6d86ea92ae2df6e89381  guix-build-d55753626292/output/x86_64-apple-darwin/bitcoin-d55753626292-x86_64-apple-darwin-unsigned.tar.gz
f014d3dba23bbd1f765b3b9541ef8af47f2d807b2e1bc56c20ccd1aa2e11d1c0  guix-build-d55753626292/output/x86_64-apple-darwin/bitcoin-d55753626292-x86_64-apple-darwin-unsigned.zip
b3e0d775e5aac539f33a1bfd39059dbb19368b38ada2c1dabf5ebd6fd782b820  guix-build-d55753626292/output/x86_64-linux-gnu/SHA256SUMS.part
ea51bd23d6e40ae306c94a04d145a2973ed888603c034c5a702942aaaba9a551  guix-build-d55753626292/output/x86_64-linux-gnu/bitcoin-d55753626292-x86_64-linux-gnu-debug.tar.gz
9c1d72510c0d78ba602ceff3b4d62455bf8757dfa28f95388834f027990f72ae  guix-build-d55753626292/output/x86_64-linux-gnu/bitcoin-d55753626292-x86_64-linux-gnu.tar.gz
c7e716768276db9227265512b73a85e5dbc3d70a9bfa05f55ce3b90047f12ca3  guix-build-d55753626292/output/x86_64-w64-mingw32/SHA256SUMS.part
ad89a26a6f61bc65ae4904f8a8ff48f4a2e62c3b49bb4e6ebfdd4a8fae8fdd61  guix-build-d55753626292/output/x86_64-w64-mingw32/bitcoin-d55753626292-win64-codesigning.tar.gz
9447f5cde1617225b65caa70d0d76faf864320d050005160bf1a69c3d0f18175  guix-build-d55753626292/output/x86_64-w64-mingw32/bitcoin-d55753626292-win64-debug.zip
0dc73ff188474f8ea27194b28c09e7e4846c13264b2ddf740152a2e4481f5662  guix-build-d55753626292/output/x86_64-w64-mingw32/bitcoin-d55753626292-win64-setup-unsigned.exe
a1dbb15ca1999363cc936e2bab3eed5c512a4b1acc5d5f97aa125e86687bfb8f  guix-build-d55753626292/output/x86_64-w64-mingw32/bitcoin-d55753626292-win64-unsigned.zip
```

hebasto reviewed (approved):
ACK d55753626292be17fbb300b54601942ff2730729.

janb84 reviewed (commented):
re ACK d55753626292be17fbb300b54601942ff2730729

change since last ack: 
- changes in comments in script.

fanquake commented:
Please rebase. Please also remove all the (what looks like AI generated) text from the commit messages & PR description. You should also post your Guix build result.

151henry151 commented on CMakeLists.txt:
```diff
@@ -619,17 +619,11 @@ if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.29)
   set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
 else()
-  set(CMAKE_SKIP_BUILD_RPATH TRUE)
+  set(CMAKE_SKIP_INSTALL_RPATH TRUE)
```
I don't think so - it looks like CMAKE_SKIP_INSTALL_RPATH was already in the file. I believe I'm just removing CMAKE_SKIP_BUILD_RPATH. The diff display seems a bit misleading here since the rebase shifted line numbers. Let me know if I'm wrong here but that's what it looks like to me.

151henry151 commented:
> Thanks. However you should do a build for all `HOSTS`. As this is changing code that effects all `HOSTS`.

Thanks for the guidance.

Guix build for 1e6aa74b4b628e8acafe6a3d2aaf2a6aa0d9d290:

046665733b06b855c98198d8fef9fe5d158f67f4dcbc15000cde0b1cc7a678d5  arm64-apple-darwin/bitcoin-1e6aa74b4b62-arm64-apple-darwin-unsigned.zip
04a35a19a5351a70e3dd66159e2cea6d8ac4a6e20f3cd0423d7b812be9b93ff5  arm64-apple-darwin/bitcoin-1e6aa74b4b62-arm64-apple-darwin-unsigned.tar.gz
09bd1e944d37759c1f52c3c0528849f6b88c3b6081f1f38592431b13237f8f63  aarch64-linux-gnu/bitcoin-1e6aa74b4b62-aarch64-linux-gnu.tar.gz
2d0ec2116b784bbbb00b5cc1cfba8a5f3ef0f07ffcb58e3c8af4c2323804057a  powerpc64-linux-gnu/bitcoin-1e6aa74b4b62-powerpc64-linux-gnu.tar.gz
3280c4015701a89883dbb40a4525608dfce00246e44f84a83de79baba45fde10  arm64-apple-darwin/bitcoin-1e6aa74b4b62-arm64-apple-darwin-codesigning.tar.gz
4f2f3d9c61fddfb4a9858375665dd72e3e4aa6a465e39f579bfea46793eec2d0  dist-archive/bitcoin-1e6aa74b4b62.tar.gz
4ffd2ac8ede8790789ee77dc2a4fe768bfb6aa4c22a19661d30e084426c7b4a9  x86_64-apple-darwin/bitcoin-1e6aa74b4b62-x86_64-apple-darwin-unsigned.tar.gz
58cb5b66a6353719708bd87262e66491d5b67a70be9445c2fa6ba53f83a266dc  x86_64-w64-mingw32/bitcoin-1e6aa74b4b62-win64-setup-unsigned.exe
60ed062fbfeb2496bb1726a1e53ad17a2a6078114c4e81f3f00499ca3ab810b4  x86_64-w64-mingw32/bitcoin-1e6aa74b4b62-win64-codesigning.tar.gz
74a9e3d88e86cbbb6cc1c6d2c3e851820a23ebae07c22f6d65d2697367e902f7  x86_64-w64-mingw32/bitcoin-1e6aa74b4b62-win64-unsigned.zip
7852ac3c67360a28522051129758ca4a9303641e3511f6714d0f738cb9a816b8  x86_64-linux-gnu/bitcoin-1e6aa74b4b62-x86_64-linux-gnu-debug.tar.gz
7aea3b0323a8bb75f4b3d16d027a8b80cb34a39a9d5617da244452a56477c0b3  arm-linux-gnueabihf/bitcoin-1e6aa74b4b62-arm-linux-gnueabihf-debug.tar.gz
7f006eb727d5d93e696522b58af0af794a273d7c803818732718036af3c9688e  riscv64-linux-gnu/bitcoin-1e6aa74b4b62-riscv64-linux-gnu.tar.gz
7fb74c5d62159a98bcb53831812b3343f2fda8dbbcc29d817cd7c4553b86e22e  x86_64-apple-darwin/bitcoin-1e6aa74b4b62-x86_64-apple-darwin-unsigned.zip
98f998c3e81d151835a51a610cf29820bf0af4096fc5e1580be10d9b750f1b3a  powerpc64-linux-gnu/bitcoin-1e6aa74b4b62-powerpc64-linux-gnu-debug.tar.gz
a4474d21f88d5725a84ca25268167978aff12c92b0da94488ff07bdbb107e8ef  x86_64-w64-mingw32/bitcoin-1e6aa74b4b62-win64-debug.zip
a98b6bee8f71f4f686b93f3358ef8ccd6492566e2eff6df391f43139fac4726c  arm-linux-gnueabihf/bitcoin-1e6aa74b4b62-arm-linux-gnueabihf.tar.gz
c7dc5634631bc2c6f3a2110b521ee44b7bde97795eb51facb8d33c9a3f999937  riscv64-linux-gnu/bitcoin-1e6aa74b4b62-riscv64-linux-gnu-debug.tar.gz
e9f2b666792a87d2f049f1c25d301d4ceba5daf75c8679b243f308b9da80959a  aarch64-linux-gnu/bitcoin-1e6aa74b4b62-aarch64-linux-gnu-debug.tar.gz
f1b1a102d01f486afd17a8c98437183f8a50bd72ea40800ad0ad4d5a4b9f0d47  x86_64-linux-gnu/bitcoin-1e6aa74b4b62-x86_64-linux-gnu.tar.gz
f761f9e63a40d13577275a98e56fa129de8049f765a0ca30959c7f880167b153  x86_64-apple-darwin/bitcoin-1e6aa74b4b62-x86_64-apple-darwin-codesigning.tar.gz

hebasto reviewed (commented):
My Guix build:
```
x86_64
a3647a46e00124b207263b695d9598cf9fe6c845ea9072c933462dd0e31a0ed9  guix-build-a02282d20c87/output/aarch64-linux-gnu/SHA256SUMS.part
6e3b9a2e39e207d98c347ccc0a81dfc6c79484befa5dbbeaaccdd0b44b759e99  guix-build-a02282d20c87/output/aarch64-linux-gnu/bitcoin-a02282d20c87-aarch64-linux-gnu-debug.tar.gz
827f5c070c2ee3c8e1eb96cfd6f05551e4a69a00e422f808c15840026c7a27fc  guix-build-a02282d20c87/output/aarch64-linux-gnu/bitcoin-a02282d20c87-aarch64-linux-gnu.tar.gz
0614c466cb38f98fa4b87fed9555d82cec60e3e46754b6bee17f00a87ef5bc74  guix-build-a02282d20c87/output/arm-linux-gnueabihf/SHA256SUMS.part
7283dfd3c64fb53ae999e19715612d027e6a5515e6d7400bd8a37f066211dd0b  guix-build-a02282d20c87/output/arm-linux-gnueabihf/bitcoin-a02282d20c87-arm-linux-gnueabihf-debug.tar.gz
2566cc5ffb73fa2eac5f18b0628502d0dbd8a87b596bd32200769e186e59158e  guix-build-a02282d20c87/output/arm-linux-gnueabihf/bitcoin-a02282d20c87-arm-linux-gnueabihf.tar.gz
8c641fbdefca6e70465685e502dae0a27f0d8ebb9d023671884fdd0982dccf62  guix-build-a02282d20c87/output/arm64-apple-darwin/SHA256SUMS.part
0bbad46352e730bd874832d44697f5f8b2b167318848a81895fa6d27bb902788  guix-build-a02282d20c87/output/arm64-apple-darwin/bitcoin-a02282d20c87-arm64-apple-darwin-codesigning.tar.gz
1b1b0f08e0e4f5dfb11aaf9a934f064ea6bae671dde55c0a6f357e08a3886953  guix-build-a02282d20c87/output/arm64-apple-darwin/bitcoin-a02282d20c87-arm64-apple-darwin-unsigned.tar.gz
5718e63cfdacc2142e6b765400c5ec694456e9d7f28528041ddc41ba0258deef  guix-build-a02282d20c87/output/arm64-apple-darwin/bitcoin-a02282d20c87-arm64-apple-darwin-unsigned.zip
79d1ff42f480bd6894b1d44fd7a59bc7c3b17cf25cccae7fe10b7f540d37af10  guix-build-a02282d20c87/output/dist-archive/bitcoin-a02282d20c87.tar.gz
a3ec88591539b7467572dfc84bdd46e5e545379b9af61b6c99e5c6c06303f032  guix-build-a02282d20c87/output/powerpc64-linux-gnu/SHA256SUMS.part
d29af70d201af3d526d50db42808c5b7b597f221892efc6269932be7f88eb92d  guix-build-a02282d20c87/output/powerpc64-linux-gnu/bitcoin-a02282d20c87-powerpc64-linux-gnu-debug.tar.gz
69f2c179d415f21e3212003085a32c25b9f584e80ff8512a108a6cfa800e0024  guix-build-a02282d20c87/output/powerpc64-linux-gnu/bitcoin-a02282d20c87-powerpc64-linux-gnu.tar.gz
f74381366c006c40f95f2b1bf67d45f8015e16080333539b238eddb7d2e41d3b  guix-build-a02282d20c87/output/riscv64-linux-gnu/SHA256SUMS.part
86606f74f26ceeff2529b4260abcbead10fda7fce3f3d0fc6226d06cff1230f9  guix-build-a02282d20c87/output/riscv64-linux-gnu/bitcoin-a02282d20c87-riscv64-linux-gnu-debug.tar.gz
c2b00950765051fde50dc2f3d7704e9134e7d777f7b18fdd0f5ca7d43c08707b  guix-build-a02282d20c87/output/riscv64-linux-gnu/bitcoin-a02282d20c87-riscv64-linux-gnu.tar.gz
3cc8638a8f32583ed396381f74e0becff6acbd9065f51ad49381da528385572f  guix-build-a02282d20c87/output/x86_64-apple-darwin/SHA256SUMS.part
e73bdd69b6dd1663bd3ead538c32d1d05c604ba00961fd5c6596785e97e1ddb8  guix-build-a02282d20c87/output/x86_64-apple-darwin/bitcoin-a02282d20c87-x86_64-apple-darwin-codesigning.tar.gz
cd0292ad0de54b9cd97680867db44748d8e4cedceb61ae1cb6669a05bf1c3cae  guix-build-a02282d20c87/output/x86_64-apple-darwin/bitcoin-a02282d20c87-x86_64-apple-darwin-unsigned.tar.gz
486ee2017e74a3b9c4128b9d313e03614b8bb64bac999fab9b600f26970f50f6  guix-build-a02282d20c87/output/x86_64-apple-darwin/bitcoin-a02282d20c87-x86_64-apple-darwin-unsigned.zip
181c8380d154d6aeab300418ffd8f6aedca852ac939a8991aa3968fb08ec5bfd  guix-build-a02282d20c87/output/x86_64-linux-gnu/SHA256SUMS.part
760b99aa489bf1cdb87b479b810478afc0e1157a129ff7ede63c531a6a3eac42  guix-build-a02282d20c87/output/x86_64-linux-gnu/bitcoin-a02282d20c87-x86_64-linux-gnu-debug.tar.gz
1748aec2dd1e54ffa07571b9d1569403b3b992972cdeedc033b280aa339e2cf8  guix-build-a02282d20c87/output/x86_64-linux-gnu/bitcoin-a02282d20c87-x86_64-linux-gnu.tar.gz
f8eaf2e58a22230b160309b478933a584bd929726cb3edc391804c916c5a4000  guix-build-a02282d20c87/output/x86_64-w64-mingw32/SHA256SUMS.part
495b7643f78e8a226687e1d8bb1a182866a725bae9854510baa3c3b883a869bd  guix-build-a02282d20c87/output/x86_64-w64-mingw32/bitcoin-a02282d20c87-win64-codesigning.tar.gz
37325cbe0ba7088f324a35b37a808b7a40cbffba3add06ba74d07203f5b6aedd  guix-build-a02282d20c87/output/x86_64-w64-mingw32/bitcoin-a02282d20c87-win64-debug.zip
54d432ad92debbd16e7b6d360af463a85d042c3e2812fd23805735a9a146f993  guix-build-a02282d20c87/output/x86_64-w64-mingw32/bitcoin-a02282d20c87-win64-setup-unsigned.exe
cf2b18d4073f6c44aca5169febc1c88cf2cc44f56e280b2030c2b96632b1dea6  guix-build-a02282d20c87/output/x86_64-w64-mingw32/bitcoin-a02282d20c87-win64-unsigned.zip
```

hebasto reviewed (approved):
ACK a02282d20c87471a3399e4061c7edad7ecdb391f.

hebasto reviewed (commented):
@151henry151 

Could you please also remove the workaround for NetBSD by reverting commit 11115e9aa845d675a88e18e729913f0aaa11e322?

I've tested it on NeBSD 10.1, and can confirm I'm sure it's no longer necessary.

151henry151 committed:
build: Remove CMAKE_SKIP_BUILD_RPATH and SKIP_BUILD_RPATH settings

Remove CMake settings that are no longer needed after reordering Guix build script to perform binary checks after installation.

Also removes unused CMake maintenance targets (check-security and check-symbols) and updates security checks to include libexec/ directory binaries (see PR #31679).

hebasto commented on CMakeLists.txt:
```diff
@@ -621,26 +621,14 @@ if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.29)
-else()
-  set(CMAKE_SKIP_BUILD_RPATH TRUE)
-endif()
+set(CMAKE_SKIP_INSTALL_RPATH TRUE)
```
Is there any reason to keep this line?

151henry151 replied:
I had thought it was still needed, but on closer analysis you're correct, thanks for catching that. Guix still passes `-DCMAKE_SKIP_RPATH=TRUE`, so the post-install security and symbol checks remain on the installed binaries without hard-coding it in the top-level CMake file, and NetBSD keeps its RPATH behavior. I've amended the commit.

hebasto reviewed (approved):
ACK 2594d5a189e52052c2019faccaa47f2affdc48e1.

My Guix build:
```
aarch64
a02136d07e0aab9f6c0352fe280226db51bb34000f6db9163a64ca7ee05902b0  guix-build-2594d5a189e5/output/aarch64-linux-gnu/SHA256SUMS.part
fddfcf1d14219e7df35dd4c25546b99df2edb1fc29856c7ae1f7a9f27bfb24b1  guix-build-2594d5a189e5/output/aarch64-linux-gnu/bitcoin-2594d5a189e5-aarch64-linux-gnu-debug.tar.gz
488d6c14892a1753495091c8a49a00d1f0e3d2702d9d7b3536fa98e714e217ea  guix-build-2594d5a189e5/output/aarch64-linux-gnu/bitcoin-2594d5a189e5-aarch64-linux-gnu.tar.gz
f8ca153fafb90d5c4a120e89fb229f38e350326ea370e530fc470a6efccb31bf  guix-build-2594d5a189e5/output/arm-linux-gnueabihf/SHA256SUMS.part
c09e05f1d64c599ceb7ba11deeebfacb90e92224901daa2d7cb519ddf7a07dcc  guix-build-2594d5a189e5/output/arm-linux-gnueabihf/bitcoin-2594d5a189e5-arm-linux-gnueabihf-debug.tar.gz
d274a9f01981b5d8b7753ac7619ee4e42da91f7e82906c17ad111d6b8a71d33a  guix-build-2594d5a189e5/output/arm-linux-gnueabihf/bitcoin-2594d5a189e5-arm-linux-gnueabihf.tar.gz
b07d53c1e2168ac09857f9b9f3403a4536a6255d7a6752cd066c965fd0891798  guix-build-2594d5a189e5/output/arm64-apple-darwin/SHA256SUMS.part
780255ae01d64925590f5282b05f5e81bb440d01955afecd4d126aeee768f2ff  guix-build-2594d5a189e5/output/arm64-apple-darwin/bitcoin-2594d5a189e5-arm64-apple-darwin-codesigning.tar.gz
b6c2391b7266c73234be419d270c2d23324814b8af3b8f43bab504eb9bebb06c  guix-build-2594d5a189e5/output/arm64-apple-darwin/bitcoin-2594d5a189e5-arm64-apple-darwin-unsigned.tar.gz
c9793c8b2289df9beff9a308e56dbb15e14a99e1fc3d281e754d056056f65801  guix-build-2594d5a189e5/output/arm64-apple-darwin/bitcoin-2594d5a189e5-arm64-apple-darwin-unsigned.zip
491fd4702c402b882e7f8da84b42e48aca5f195c7163422323cbde4b9f432ab9  guix-build-2594d5a189e5/output/dist-archive/bitcoin-2594d5a189e5.tar.gz
1cb42ab34a36ebcf4e5c2a733329c27924dc6e005b394a2c9d305a7522c588d9  guix-build-2594d5a189e5/output/powerpc64-linux-gnu/SHA256SUMS.part
f68f05109688fb974aa40287d370a437f20e1f26c529731b5509eac59777009c  guix-build-2594d5a189e5/output/powerpc64-linux-gnu/bitcoin-2594d5a189e5-powerpc64-linux-gnu-debug.tar.gz
0e7435455372d76b1a94ad7a3a046f9ffc4b8f6e6e44ebe5ccfe74c1d004d550  guix-build-2594d5a189e5/output/powerpc64-linux-gnu/bitcoin-2594d5a189e5-powerpc64-linux-gnu.tar.gz
91d53dd08d5b5b1d891ed357446a573275bafcd75888bb9e0a6567d9da1a61d4  guix-build-2594d5a189e5/output/riscv64-linux-gnu/SHA256SUMS.part
8ae470b91a8f676bffbdbe23d0849a417a086e3152cf26d422b990f925f58dfc  guix-build-2594d5a189e5/output/riscv64-linux-gnu/bitcoin-2594d5a189e5-riscv64-linux-gnu-debug.tar.gz
818917545a14983fc9221a1d6c0f0193a3bd4d0f2d6159284de955323984785c  guix-build-2594d5a189e5/output/riscv64-linux-gnu/bitcoin-2594d5a189e5-riscv64-linux-gnu.tar.gz
a72b04cf8760bfe054461ed7bb04099883328fbef940324e1ddf9c6a09150265  guix-build-2594d5a189e5/output/x86_64-apple-darwin/SHA256SUMS.part
b27fdc8e873aba8fb92de5d7f214a44abc3c6c2cf2f92774424f9e1e24e143f5  guix-build-2594d5a189e5/output/x86_64-apple-darwin/bitcoin-2594d5a189e5-x86_64-apple-darwin-codesigning.tar.gz
2f6657d3c9663ae1d424ecc9def1069ca430358eb667ee5c5c4b2962ac33e569  guix-build-2594d5a189e5/output/x86_64-apple-darwin/bitcoin-2594d5a189e5-x86_64-apple-darwin-unsigned.tar.gz
090f4621f9aef54dae89ff1d1c05e918a169efd9c74da7b0f7ae1864c5dc3266  guix-build-2594d5a189e5/output/x86_64-apple-darwin/bitcoin-2594d5a189e5-x86_64-apple-darwin-unsigned.zip
b1acda62e967e7cd487547d3e81b63e13efdcad4248202b98de363d313b24381  guix-build-2594d5a189e5/output/x86_64-linux-gnu/SHA256SUMS.part
efddeb0fe68f73de792c63832379e59763fba8bc1d10ac7bf8d3a2a840a1d56e  guix-build-2594d5a189e5/output/x86_64-linux-gnu/bitcoin-2594d5a189e5-x86_64-linux-gnu-debug.tar.gz
ceedded0f573ebffd4d66e880658ac3be0c955edced38467639e4e5dce2c5aeb  guix-build-2594d5a189e5/output/x86_64-linux-gnu/bitcoin-2594d5a189e5-x86_64-linux-gnu.tar.gz
dfafeb32dcef307ceac8b08f0ed06439df8aa8c3710f50e42d974f2364116d12  guix-build-2594d5a189e5/output/x86_64-w64-mingw32/SHA256SUMS.part
c3089ac2e7c85cbd77f30aa3fc3b30ac7cfa2e72bcd4052306ba6c541ec5eb6a  guix-build-2594d5a189e5/output/x86_64-w64-mingw32/bitcoin-2594d5a189e5-win64-codesigning.tar.gz
1d145af4b3acc8aef7a92c6f686710235bec0bc786d19ac071e7cec533cdf87b  guix-build-2594d5a189e5/output/x86_64-w64-mingw32/bitcoin-2594d5a189e5-win64-debug.zip
82115a95c29d083200f10815d220380403fdfcc17b611a37b1d2765a2fe79710  guix-build-2594d5a189e5/output/x86_64-w64-mingw32/bitcoin-2594d5a189e5-win64-setup-unsigned.exe
4468f0ee9eef439d107240466310d94ec8b3dc472b2b424da73d1ef0a50e3148  guix-build-2594d5a189e5/output/x86_64-w64-mingw32/bitcoin-2594d5a189e5-win64-unsigned.zip
```

hebasto commented:
https://github.com/bitcoin/bitcoin/pull/33247#issuecomment-3534660813

> My guix build hashes:

@151henry151 

To make hashes easily comparable, they are sorted using `env LC_ALL=C sort`. The full command to retrieve the hashes is as follows:
```
find guix-build-$(git rev-parse --short=12 HEAD)/output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum
```

fanquake commented on contrib/guix/libexec/build.sh:
```diff
@@ -282,6 +277,13 @@ mkdir -p "$DISTSRC"
+    python3 "${DISTSRC}/contrib/guix/security-check.py" "${INSTALLPATH}/bin/"* "${INSTALLPATH}/libexec/"*
+    # Check that executables only contain allowed version symbols.
+    echo "Running symbol and dynamic library checks on installed executables..."
+    python3 "${DISTSRC}/contrib/guix/symbol-check.py" "${INSTALLPATH}/bin/"* "${INSTALLPATH}/libexec/"*
```
```bash
/distsrc-base/distsrc-275349011d15-x86_64-linux-gnu/installed/bitcoin-275349011d15/bin/bitcoin
/distsrc-base/distsrc-275349011d15-x86_64-linux-gnu/installed/bitcoin-275349011d15/bin/bitcoin-cli
/distsrc-base/distsrc-275349011d15-x86_64-linux-gnu/installed/bitcoin-275349011d15/bin/bitcoin-qt
/distsrc-base/distsrc-275349011d15-x86_64-linux-gnu/installed/bitcoin-275349011d15/bin/bitcoin-tx
/distsrc-base/distsrc-275349011d15-x86_64-linux-gnu/installed/bitcoin-275349011d15/bin/bitcoin-util
/distsrc-base/distsrc-275349011d15-x86_64-linux-gnu/installed/bitcoin-275349011d15/bin/bitcoin-wallet
/distsrc-base/distsrc-275349011d15-x86_64-linux-gnu/installed/bitcoin-275349011d15/bin/bitcoind
/distsrc-base/distsrc-275349011d15-x86_64-linux-gnu/installed/bitcoin-275349011d15/libexec/bitcoin-gui
/distsrc-base/distsrc-275349011d15-x86_64-linux-gnu/installed/bitcoin-275349011d15/libexec/bitcoin-node
/distsrc-base/distsrc-275349011d15-x86_64-linux-gnu/installed/bitcoin-275349011d15/libexec/test_bitcoin
```
